### PR TITLE
*: add the traffic statistic

### PR DESCRIPTION
**Pull request by nolouch**
ref https://github.com/pingcap/tidb/issues/57543

### CODE REVIEW

File: internal/client/metrics_collector.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		size = req.Cop().Size()
+	case tikvrpc.CmdPrewrite:
+		size = req.Prewrite().Size()
+	case tikvrpc.CmdCommit:
+		size = req.Commit().Size()
+	case tikvrpc.CmdPessimisticLock:
+		size = req.PessimisticLock().Size()
+	case tikvrpc.CmdMPPTask:
+		size = req.DispatchMPPTask().Size()
+	default:
```
Comment (crazycs520) [inline internal/client/metrics_collector.go:105]: also need record other request?

```
CmdPessimisticRollback
CmdBatchRollback
CmdFlush
CmdResolveLock
CmdCheckTxnStatus
CmdCheckSecondaryLocks
CmdScanLock
CmdTxnHeartBeat
CmdGetHealthFeedback
```

File: internal/locate/region_request.go [hunk 2]
```diff
@@ -844,13 +842,19 @@ func (s *RegionRequestSender) SendReqCtx(
 			resp, err = tikvrpc.GenRegionErrorResp(req, &errorpb.Error{EpochNotMatch: &errorpb.EpochNotMatch{}})
 			return resp, nil, retryTimes, err
 		}
-
-		var isLocalTraffic bool
-		if staleReadCollector != nil && s.replicaSelector != nil && s.replicaSelector.target != nil {
-			isLocalTraffic = s.replicaSelector.target.store.IsLabelsMatch(s.replicaSelector.option.labels)
-			staleReadCollector.onReq(req, isLocalTraffic)
+		// patch the access location if it is not set under region request sender. which inculdes the coprocessor,
```
Comment (bufferflies) [inline internal/locate/region_request.go:845]:
```suggestion
		// patch the access location if it is not set under region request sender. which includes the coprocessor,
```

Comment (bufferflies) [reply internal/locate/region_request.go:845]: how about use one function to get the access zone from s?

**Review by bufferflies — COMMENTED** [review]
rest lgtm

File: internal/client/metrics_collector.go (hunk 1 above)
Comment (nolouch) [reply internal/client/metrics_collector.go:75]: added some relative to the user app.

**Review by crazycs520 — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]

**Pull request merged**